wpilibc/src/main/native/cpp/ADIS16448_IMU.cpp

```diff
@@ -507,6 +509,25 @@ void ADIS16448_IMU::Reset() {
   m_integ_gyro_angle_x = 0.0;
   m_integ_gyro_angle_y = 0.0;
   m_integ_gyro_angle_z = 0.0;
+  angleOffset = frc::Rotation3d{};
+}
+
+void ADIS16448_IMU::Reset(Rotation3d angle) {
+  std::scoped_lock sync(m_mutex);
+  m_integ_gyro_angle_x = 0.0;
+  m_integ_gyro_angle_y = 0.0;
+  m_integ_gyro_angle_z = 0.0;
+  angleOffset = -angle;
+
+  if (m_simGyroAngleX) {
+    m_simGyroAngleX.Set(0.0);
```

Comment:
I don't think this would cause the tests to fail (since the `SimDouble`s should have a value of 0 anyways), but note that this was created as an input to the gyro:
```cpp
    m_simGyroAngleX =
        m_simDevice.CreateDouble("gyro_angle_x", hal::SimDevice::kInput, 0.0);
```